src/main/java/com/aliyun/hitsdb/client/TSDBClient.java

```diff
@@ -134,18 +138,41 @@ public TSDBClient(Config config) throws HttpClientInitException {
             }
             throw new RuntimeException(e);
         }
+
+        if (config.isDnsCheckEnable()) {
+            DNSChecker dnsChecker = new DNSChecker(this, config.getHost(), config.getDnsCheckInterval());
+            dnsChecker.start();
+            LOGGER.info("dns checker start");
+        }
     }
 
     private static final String EMPTY_HOLDER = new JSONObject().toJSONString();
     private static final String VIP_API = "/api/vip_health";
 
     private void checkConnection() {
-        httpclient.post(VIP_API, EMPTY_HOLDER);
+        getHttpclient().post(VIP_API, EMPTY_HOLDER);
         if (secondaryClient != null) {
             secondaryClient.post(VIP_API, EMPTY_HOLDER);
         }
     }
 
+    void resetAllConnections() {
+        LOGGER.info("start reset all connections");
+        try {
+            getHttpclient().close(true);
+        } catch (Exception e) {
+            LOGGER.warn("close getHttpclient() error ", e);
+        }
+
+        this.httpclient.set(HttpClientFactory.createHttpClient(config));
+        getHttpclient().start();
```

Comment:
要么在这里的 `getHttpclient().start()` 中加异常处理，要么在DNSCheck的常驻线程中加异常处理。
否则如果这里万一有未捕获的异常抛出，将会导致这个定时任务中止，后面也不再执行。

该问题在XX局点的应用中出现过，JDK官方对于定时任务的异常有下述描述：

> If any execution of the task encounters an exception, subsequent executions are suppressed.Otherwise, the task will only terminate via cancellation or termination of the executor.

Reply:
DNSChecker中已经加入了异常处理